Signal/src/views/OWSFlatButton.swift

```diff
@@ -0,0 +1,176 @@
+                        font: font,
+                        titleColor: titleColor )
+        button.setBackgroundColors(backgroundColor)
+        button.useDefaultCornerRadius()
+        button.addTarget(target:target, selector:selector)
+        return button
+    }
+
+    public class func fontForHeight(_ height: CGFloat) -> UIFont {
+        let fontPointSize = round(height * 0.45)
+        return UIFont.ows_mediumFont(withSize:fontPointSize)!
+    }
+
+    // MARK: Methods
+
+    public func setTitle(title: String, font: UIFont,
+                         titleColor: UIColor ) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.setTitle(title, for: .normal)
+        button.setTitleColor(titleColor, for: .normal)
+        button.titleLabel!.font = font
+    }
+
+    public func setBackgroundColors(upColor: UIColor,
+                                    downColor: UIColor ) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.setBackgroundImage(UIImage(color:upColor), for: .normal)
+        button.setBackgroundImage(UIImage(color:downColor), for: .highlighted)
+    }
+
+    public func setBackgroundColors(_ backgroundColor: UIColor ) {
+        setBackgroundColors(upColor: backgroundColor,
+                            downColor: backgroundColor.withAlphaComponent(0.7) )
+    }
+
+    public func setSize(width: CGFloat, height: CGFloat) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.autoSetDimension(.width, toSize:width)
+        button.autoSetDimension(.height, toSize:height)
+    }
+
+    public func useDefaultCornerRadius() {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        // To my eye, this radius tends to look right regardless of button size
+        // (within reason) or device size. 
+        button.layer.cornerRadius = 5
+        button.clipsToBounds = true
+    }
+
+    public func setEnabled(_ isEnabled: Bool) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.isEnabled = isEnabled
+    }
+
+    public func addTarget(target:Any,
+                          selector: Selector) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.addTarget(target, action:selector, for:.touchUpInside)
+    }
+
+    public func setPressedBlock(_ pressedBlock: @escaping () -> Void) {
+        guard self.pressedBlock == nil else {
+            owsFail("Button already has pressed block.")
+            return
+        }
+        self.pressedBlock = pressedBlock
+    }
+
+    internal func buttonPressed() {
+        guard let pressedBlock = pressedBlock else {
```

Comment:
nit: This method could just be `pressedBlock?()`

Reply:
Done.